controller_manager:
  ros__parameters:
    update_rate: 500  # Hz

    joint_state_broadcaster:
      type: joint_state_broadcaster/JointStateBroadcaster

    pid_controller:
      type: pid_controller/PidController

pid_controller:
  ros__parameters:
    update_rate: 100 # Hz

    dof_names:
      - left_hip_pitch_joint
      - left_hip_roll_joint
      - left_hip_yaw_joint
      - left_knee_joint
      - left_ankle_pitch_joint
      - left_ankle_roll_joint
      - right_hip_pitch_joint
      - right_hip_roll_joint
      - right_hip_yaw_joint
      - right_knee_joint
      - right_ankle_pitch_joint
      - right_ankle_roll_joint
      - waist_yaw_joint
      - left_shoulder_pitch_joint
      - left_shoulder_roll_joint
      - left_shoulder_yaw_joint
      - left_elbow_joint
      - left_wrist_roll_joint
      - left_wrist_pitch_joint
      - left_wrist_yaw_joint
      - right_shoulder_pitch_joint
      - right_shoulder_roll_joint
      - right_shoulder_yaw_joint
      - right_elbow_joint
      - right_wrist_roll_joint
      - right_wrist_pitch_joint
      - right_wrist_yaw_joint
    
    command_interface: effort

    reference_and_state_interfaces: ["position", "velocity"]

    gains:
      left_hip_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_hip_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_hip_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_knee_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_ankle_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_ankle_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_hip_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_hip_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_hip_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_knee_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_ankle_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_ankle_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      waist_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_shoulder_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_shoulder_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_shoulder_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_elbow_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_wrist_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_wrist_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      left_wrist_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_shoulder_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_shoulder_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_shoulder_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_elbow_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_wrist_roll_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_wrist_pitch_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      right_wrist_yaw_joint: {p: 10.0, i: 0.0, d: 0.0, i_clamp_max: 5.0, i_clamp_min: -5.0, feedforward_gain: 0.0}
      